lib/util/SortableSet.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+module.exports = class SortableSet extends Set {
+
+	constructor(initialIterable, defaultSort) {
+		super(initialIterable);
+		this._sortFn = defaultSort;
+		this._lastActiveSortFn = null;
+		this._isSorted = false;
+	}
+
+	/**
+	 * @param {any} value - value to add to set
+	 * @returns {SortableSet} - returns itself
+	 */
+	add(value) {
+		this._lastActiveSortFn = null;
+		this._isSorted = false;
+		super.add(value);
+		return this;
+	}
+
+	/**
+	 * @returns {void}
+	 */
+	clear() {
+		this._lastActiveSortFn = null;
+		this._isSorted = false;
+		super.clear();
+	}
+
+	/**
+	 * @param {Function} sortFn - function to sort the set
+	 * @returns {void}
+	 */
+	sortWith(sortFn) {
+		if(this._isSorted && sortFn === this._lastActiveSortFn) {
```

Comment:
We could skip sorting if `this.size === 0`. This avoids creating a new empty array below.